systest/bgindex/parallel_test.go

```diff
@@ -96,24 +102,35 @@ func TestParallelIndexing(t *testing.T) {
 	}
 
 	fmt.Println("building indexes in background for int and string data")
-	if err := dg.Alter(context.Background(), &api.Operation{
-		Schema: `
+	// Wait until previous indexing is complete.
+	for {
```

Comment:
This will cause the test to be hung forever if there is some issue. Better add `RetryUntilSuccess` here as well so that it is easy to figure out what was the error after `n` tries.

Reply:
This will only retry if the error is of type Indexing in Progress in all other cases it will fail. That part is fine i believe. Indexing can take much longer to complete especially when race is enabled. Anyways we have checks in place to timeout tests itself after 10 minutes so that will take care of this part

Reply:
Hmm. Makes sense